mempool/cat/store.go

```diff
@@ -165,4 +176,41 @@ func (s *store) reset() {
 	defer s.mtx.Unlock()
 	s.bytes = 0
 	s.txs = make(map[types.TxKey]*wrappedTx)
+	s.orderedTxs = make([]*wrappedTx, 0)
+}
+
+func (s *store) orderTx(tx *wrappedTx) {
+	idx := s.getTxOrder(tx)
+	s.orderedTxs = append(s.orderedTxs[:idx], append([]*wrappedTx{tx}, s.orderedTxs[idx:]...)...)
+}
+
+func (s *store) deleteOrderedTx(tx *wrappedTx) error {
+	if len(s.orderedTxs) == 0 {
+		return fmt.Errorf("ordered transactions list is empty")
+	}
+	idx := s.getTxOrder(tx) - 1
+	if idx >= len(s.orderedTxs) || s.orderedTxs[idx] != tx {
+		return fmt.Errorf("transaction %X not found in ordered list", tx.key)
+	}
+	s.orderedTxs = append(s.orderedTxs[:idx], s.orderedTxs[idx+1:]...)
```

Comment:
if the idx is the last transaction in the ordered list, I expect `s.orderedTxs[idx+1:]` to fail.

Perhaps this needs a unit test where `deleteOrderedTx` is invoked on the last tx in the ordered list to verify there is no bug here.

Reply:
`TestStoreSimple` actually tests this by adding a tx and removing it. Since it's the only transaction, it is defacto the last tx in the ordered list